app/src/main/java/it/niedermann/owncloud/notes/android/activity/EditNoteActivity.java

```diff
@@ -75,6 +75,27 @@ protected void onCreate(final Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        setIntent(intent);
+    }
+
+    @Override
+    protected void onResume() {
+        Intent intent = getIntent();
+        DBNote note = (DBNote) intent.getSerializableExtra(PARAM_NOTE);
+
+        // Does the note retrieved from the intent match the note currently
+        // being displayed by the app
+        if ((note.getId()) != fragment.getNote().getId()) {
+            db.updateNoteAndSync(fragment.getNote(), fragment.getNote().getContent(), null);
```

Comment:
I don't think that you need this, since your `NoteEditFragment.getNote()` saves the note automatically, when called (I don't like this, but that's how it's currently works).